src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java

```diff
@@ -226,6 +227,21 @@ public String getLog(String appId) {
 		return logAppender.toString();
 	}
 
+	@Override
+	public void scale(String appId, int desiredCount) {
+		if (this.client.apps().statefulSets().withName(appId).get() != null) {
+			logger.debug(String.format("Scale a stateful app: %s to: %s", appId, desiredCount));
+			this.client.apps().statefulSets().withName(appId).scale(desiredCount, true);
```

Comment:
might be nit picky but could probably do a local:

`ScalableResource scalableResource = null;`

and assign the return value of the `withName` calls, then after the conditional block just call `scalableResource.scale(desiredCount, true)` 

?

Reply:
ok I will try something along those lines